### PR TITLE
Assert inPosition in bounds

### PR DESCRIPTION
**Pull request by AnHeuermann**
### Related Issues

List.split can fail without giving an error message in case `inPosition` is out of bounds.
This is the error that is thrown in https://github.com/OpenModelica/OpenModelica/issues/11083.

### Purpose

  - Check that List.split is in bounds or throw an error with error message.

### CODE REVIEW

File: OMCompiler/Compiler/Util/List.mo [hunk 1]
```diff
@@ -1240,6 +1241,12 @@ protected
   list<T> l1 = {}, l2 = inList;
   T e;
 algorithm
+  // Check if inPosition is a valid value
+  Error.assertion(
+    inPosition >= 0 and inPosition <= listLength(inList),
+    getInstanceName() + " out of bounds error for input argument inPosition: " + intString(inPosition) + "\n",
+    sourceInfo());
+
   true := inPosition >= 0;
```
Comment (phannebohm) [inline OMCompiler/Compiler/Util/List.mo:1250]: I think you can skip this check then.
```suggestion
```

**Comment by perost** [comment]
I'm not sure this is a good idea, none of the other `List` functions print any error messages and just using `print` also makes it impossible to suppress the error message. Since MetaModelica has fail semantics it's not necessarily a hard error if `List.split` fails, so in my opinion it should be up to the caller to determine how to handle failures.

**Comment by AnHeuermann** [comment]
I think failing without an error message makes it hard to track down issues like the one from https://github.com/OpenModelica/OpenModelica/issues/11083.

> so in my opinion it should be up to the caller to determine how to handle failures.

Maybe. Then trowing an error from Error.mo would be a solution, but that can't be used for the bootstrapping compiler.

Should we revert this commit?

**Comment by perost** [comment]
> I think failing without an error message makes it hard to track down issues like the one from #11083.
> 
> > so in my opinion it should be up to the caller to determine how to handle failures.
> 
> Maybe. Then trowing an error from Error.mo would be a solution, but that can't be used for the bootstrapping compiler.
> 
> Should we revert this commit?

I agree that it makes it harder to find errors, but that's more an issue with MetaModelica as a whole rather than only the `List` functions. Personally I prefer having the `List` functions be consistent, and none of the other functions print any errors.

`print`:ing errors is also bad since we have a lot of code where failure is expected but we don't want to print any error messages, but this is going to circumvent that and print errors anyway. So not print an error message would be the lesser of two evils in my opinion.

**Pull request merged**